client/src/components/Panels/Common/ToolAdvancedSearch.vue

```diff
@@ -0,0 +1,219 @@
+        help: {
+            type: String,
+            default: "",
+        },
+    },
+    data() {
+        return {
+            offset: 0,
+        };
+    },
+    computed: {
+        filterSettings() {
+            const newFilterSettings = {};
+            Object.entries(this.$props).forEach(([filter, value]) => {
+                if (value && value !== "") {
+                    if (filter === "panelSectionName") {
+                        newFilterSettings["panel_section_name"] = value;
+                    } else if (filter !== "showHelp") {
+                        newFilterSettings[filter] = value;
+                    }
+                }
+            });
+            return newFilterSettings;
+        },
+        hasFilters() {
+            return Object.keys(this.filterSettings).length;
+        },
+    },
+    beforeDestroy() {
+        document.querySelector(".center-panel").removeEventListener("scroll", this.onScroll, true);
+    },
+    mounted() {
+        document.querySelector(".center-panel").addEventListener("scroll", this.onScroll, true);
+    },
+    methods: {
+        onScroll(e) {
+            this.offset = e.target.scrollTop;
+        },
+        hasHelp(help) {
+            return help && help !== "\n";
+        },
+        onClear() {
+            this.$router.push({ path: "/tools/advanced_search" });
+        },
+        onOpen(tool) {
+            if (tool.id === "upload1") {
+                openGlobalUploadModal();
+            } else if (tool.form_style === "regular") {
+                const Galaxy = getGalaxyInstance();
+                // encode spaces in tool.id
+                const toolId = tool.id;
+                const toolVersion = tool.version;
+                Galaxy.router.push(`/?tool_id=${encodeURIComponent(toolId)}&version=${toolVersion}`);
+            }
+        },
+        scrollToTop() {
+            document.querySelector(".center-panel").scrollTop = 0;
+        },
+        helpSummary(help) {
+            const parser = new DOMParser();
+            const helpDoc = parser.parseFromString(help, "text/html");
+            const xpaths = [
+                "//strong[text()='What it does']/../following-sibling::*",
+                "//strong[text()='What It Does']/../following-sibling::*",
+                "//h1[text()='Synopsis']/following-sibling::*",
+                "//strong[text()='Syntax']/../following-sibling::*",
+            ];
+            const matches = [];
+            xpaths.forEach((xpath) => {
+                matches.push(
+                    helpDoc.evaluate(xpath, helpDoc, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue
+                );
+            });
+            let returnVal = null;
+            matches.forEach((match) => {
+                if (match) {
+                    returnVal = match.innerHTML + "\n";
+                }
+            });
+            return returnVal;
+        },
+    },
+};
+</script>
+
+<style>
+.btn-back-to-top {
```

Comment:
<img width="126" alt="image" src="https://user-images.githubusercontent.com/2105447/186536020-3a21790b-bf85-4fd6-bdb3-4a4ad73ea368.png">

Reply:
This button only appears when you scroll down a bit, would it be fine to let it hover over other content as many `Scroll To Top` buttons do?